File: src/algorithm/2.2.c
struct stack {
    char box[smax+1]
    int top;
}

void initialize(struct stack *s){
    s->top = 0;
}

void push(struct stack *s, cahr item){
    s->box[++s->top] = item;
}


void pop(struct stack *s){
    --s -> top;
}

int empty(struct stack *s){
    return(s->top = 0);
}

char top(struct stack *s){
    return(s->box[s->top]);
}

File: src/algorithm/2.2.ts
const smax = 100;

class Stack {
    private box: string[] = [];
    private top: number = 0;

    push(item: string): void {
        if (this.top >= smax) throw new Error("Stack overflow");
        this.box[++this.top] = item; // 1番目から使用
    }

    pop(): string {
        if (this.empty()) throw new Error("Stack underflow");
        return this.box[this.top--]; // topを返してからデクリメント
    }

    empty(): boolean {
        return this.top === 0; // top=0 なら空
    }

    peek(): string {
        if (this.empty()) throw new Error("Stack is empty");
        return this.box[this.top]; // topの要素を返す
    }
}
